### PR TITLE
[FEATURE] Introduce VCS template provider

### PR DESCRIPTION
**Pull request by eliashaeussler**
This PR adds a new template provider for VCS repositories. It can be selected as third option when running the project builder.

ToDo:

- [x] Verify handling of additional repositories (to be discussed with @mteu)
- [x] Tests
- [x] Documentation

Resolves: #51

### CODE REVIEW

**Comment by mteu** [comment]
Let's also talk about whether or not it would make sense to allow branches from a VCS provider knowing that I'm obviously thinking `git` right now.

**Comment by mteu** [comment]
@eliashaeussler Thanks, this could come in very handy to many users. Review-wise everything looks very good to me until here. ✅ Let's continue on these open checkboxes.

**Review by mteu — APPROVED** [review]
We're going to have to tackle the documentation anyhow, so I'd like to approve the PR already. Thanks for introducing the `VcsProvider` which should be helpful for a broader audience! 💛 

Acknowledging the imminent changes tackling https://github.com/CPS-IT/project-builder/issues/57 we should be fine here. 👍

**Comment by eliashaeussler** [comment]
There are still more tests needed. On it.

File: tests/src/Template/Provider/VcsProviderTest.php [hunk 1]
```diff
@@ -3,13 +3,13 @@
 declare(strict_types=1);
 
 /*
- * This file is part of the TYPO3 CMS extension "cpsit/project-builder".
+ * This file is part of the Composer package "cpsit/project-builder".
```
Comment (mteu) [inline tests/src/Template/Provider/VcsProviderTest.php:6]: Darn it. That's the result of a very frequently used snippet in my IDE. Apologies!

File: tests/src/Template/Provider/VcsProviderTest.php [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
  *
  * Copyright (C) 2022 Martin Adler <m.adler@familie-redlich.de>
  *
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 2 of the License, or
+ * the Free Software Foundation, either version 3 of the License, or
```
Comment (mteu) [inline tests/src/Template/Provider/VcsProviderTest.php:12]: 👍

**Review by mteu — APPROVED** [review]

**Pull request merged**